pkg/freezer/factory.go

```diff
@@ -0,0 +1,37 @@
+package freezer
+
+import (
+	"fmt"
+
+	"github.com/julz/freeze-proxy/pkg/daemon"
+	"github.com/julz/freeze-proxy/pkg/freezer/containerd"
+	docker "github.com/julz/freeze-proxy/pkg/freezer/docker"
+)
+
+const (
+	RuntimeTypeDocker     string = "docker"
+	RuntimeTypeContainerd string = "contaienrd"
+	RuntimeTypeCriO       string = "cri-o"
+)
+
+func GetFreezer(runtimeType string) (daemon.Freezer, error) {
```

Comment:
I wonder if we could drop this file and just put the switch in main.go. Having a factory feels a bit... java-ish ;)

Reply:
A little different idea, putting `switch in main.go` looks a bit messy... Or just rename the factory to `freeze.go` in package `/pkg/freeze/` as abstraction of two implementation methods？

Reply:
I mean it's less about the name of the file and more that we don't really need the extra abstraction/indirection here (for a start it's pretty odd/confusing for a Get method to call New..). We can always add a factory later if/when we need it.

Since in main.go there'd only need to be one switch, it should also end up only needing roughly half the lines of code to do it there.

Reply:
@julz updated